mobile/src/components/common/search/search.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC } from 'react';
+import { ImageURISource, TextInput, TouchableOpacity } from 'react-native';
+
+import { AppColor } from '~/common/enums/enums';
+import { Image, View } from '~/components/common/common';
+import { useEffect, useState } from '~/hooks/hooks';
+
+import { styles } from './styles';
+
+type Props = {
+  onSearch: (text: string) => void;
+  onVoice?: () => Promise<string>;
+};
+
+type IconSize = {
+  width: number;
+  height: number;
+};
+
+type IconName = 'search' | 'voice';
+
+const Search: FC<Props> = ({ onSearch, onVoice }) => {
+  const [text, setText] = useState('');
+  const [borderColor, setBorderColor] = useState('transparent');
+
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
+      onSearch(text);
+    }, 1000);
+
+    return () => clearTimeout(delayDebounceFn);
+  }, [text]);
+
+  const handleChangeText = (value: string): void => {
+    setText(value);
+  };
+
+  const handleVoicePress = async (): Promise<void> => {
+    if (onVoice) {
+      handleChangeText(await onVoice());
+    }
+  };
+
+  const getUriSource = (name: IconName): ImageURISource => {
+    if (name === 'search') {
+      return require('~/assets/images/search.png');
+    }
+
+    return require('~/assets/images/voice.png');
+  };
+
+  const getIcon = (name: IconName, size: IconSize): JSX.Element => {
+    return (
+      <Image
+        source={Image.resolveAssetSource(getUriSource(name))}
+        style={{ width: size.width, height: size.height }}
+      />
+    );
+  };
+
+  return (
+    <View style={{ ...styles.searchBar, borderColor: borderColor }}>
+      {getIcon('search', { width: 16, height: 16 })}
+      <TextInput
```

Comment:
All inputs should be controlled via useAppForm

Reply:
@Liza-Veis 
Please correct me if I'm wrong.
We use useAppForm to validate the input and what to validate with the search component. I don't understand how to apply useAppForm to this component. It doesn't even have a button for handle submit.

Reply:
Not only to validate, to control too

Reply:
Validation schema is optional

Reply:
@Liza-Veis mean that we don\`t need this `const [text, setText] = useState('');`, use form for store and handle input value

Reply:
We should useAppForm for all inputs in our app, it's not only for validation, we need to use it for controlling state here